org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/quickfix/DiagnosticResolution.java

```diff
@@ -139,7 +134,10 @@ public WorkspaceEdit apply() {
 		}
 	}
 
-	void configure(Options options) {
-		this.options = options;
+	void configure(Options options, Diagnostic diagnostic) {
+		this.resource = options.getResource();
```

Comment:
Why did you decide to inline the options' properties here?

Reply:
I think passing the Options as part of the API in the configure is not good, it has some information which cannot be used as part of the diagnostic resolution (like the list of diagnostics that we where using before). I did not want to change the public API if not needed, so I decided to inline the parameters that are used, hopefully next time something must be inlined, some thinking is done about if it is correct to access the data at this place or not.

Reply:
Agreed. Makes sense.